go/enclave/storage/enclavedb/event_filter.go

```diff
@@ -0,0 +1,530 @@
+	logsQuery := " et.event_sig, t1.topic, t2.topic, t3.topic, datablob, log_idx, b.hash, b.height, curr_tx.hash, curr_tx.idx, c.address "
+	receiptQuery := " rec.post_state, rec.status, rec.gas_used, rec.effective_gas_price, rec.created_contract_address, tx_sender.address, tx_contr.address, curr_tx.type "
+
+	query := "select " + logsQuery
+	if withReceipts {
+		query += "," + receiptQuery
+	}
+	query += baseReceiptJoin
+	query += baseEventJoin
+
+	var queryParams []any
+
+	if requestingAccountId != nil {
+		// Add log visibility rules
+		logsVisibQuery, logsVisibParams := logsVisibilityQuery(*requestingAccountId, withReceipts)
+		query += logsVisibQuery
+		queryParams = append(queryParams, logsVisibParams...)
+
+		// add receipt visibility rules
+		if withReceipts {
+			receiptsVisibQuery, receiptsVisibParams := receiptsVisibilityQuery(*requestingAccountId)
+			query += receiptsVisibQuery
+			queryParams = append(queryParams, receiptsVisibParams...)
+		}
+	}
+
+	query += whereCondition
+	queryParams = append(queryParams, whereParams...)
+
+	if withReceipts && requestingAccountId != nil {
+		// there is a corner case when a receipt has logs, but none are visible to the requester
+		query += " UNION ALL "
+		query += " select null, null, null, null, null, null, b.hash, b.height, curr_tx.hash, curr_tx.idx, null, " + receiptQuery
+		query += baseReceiptJoin
+		query += " where b.is_canonical=true "
+		query += " AND tx_sender.id = ? "
+		queryParams = append(queryParams, requestingAccountId)
+		query += whereCondition
+		queryParams = append(queryParams, whereParams...)
+	}
+
+	stmt, err := stmtCache.GetOrPrepare(query)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not prepare query: %w", err)
+	}
+
+	rows, err := stmt.QueryContext(ctx, queryParams...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not execute query: %w", err)
+	}
+	defer rows.Close()
+
+	receipts := make([]*core.InternalReceipt, 0)
+	logList := make([]*types.Log, 0)
+
+	empty := true
+	for rows.Next() {
+		empty = false
+		r, l, err := onRowWithEventLogAndReceipt(rows, withReceipts)
+		if err != nil {
+			return nil, nil, err
+		}
+		receipts = append(receipts, r)
+		logList = append(logList, l)
+	}
+	if rows.Err() != nil {
+		return nil, nil, rows.Err()
+	}
+
+	if withReceipts {
+		if empty {
+			return nil, nil, errutil.ErrNotFound
+		}
+		// group the logs manually to avoid complicating the db indexes
+		result := groupReceiptAndLogs(receipts, logList)
+		return result, nil, nil
+	}
+	return nil, logList, nil
+}
+
+func groupReceiptAndLogs(receipts []*core.InternalReceipt, logs []*types.Log) []*core.InternalReceipt {
+	recMap := make(map[gethcommon.Hash]*core.InternalReceipt)
+	logMap := make(map[gethcommon.Hash][]*types.Log)
+	for _, r := range receipts {
+		recMap[r.TxHash] = r
+	}
+	for _, log := range logs {
+		logList := logMap[log.TxHash]
+		if logList == nil {
```

Comment:
Minor thing but you don't need to bother with this check, just `logMap[log.TxHash] = append(logList, log)` will work because append(nil, someItem) creates a new list automatically, it handles nil list.

Reply:
this has been like this forerver. I just moved it to a new file, as the old one was getting huge